.circleci/scripts/binary_linux_test.sh

```diff
@@ -54,6 +54,13 @@ fi
 
 # Test the package
 /builder/check_binary.sh
+
+# Run MyPy (TODO: move this to builder/binary_run_in_docker.sh)
+# TODO: installing mypy isn't needed after the Docker images are updated with it
+if [[ "$PACKAGE_TYPE" != libtorch ]]; then
+  retry pip install -q mypy==0.770
+  mypy
+fi
```

Comment:
Okay this is in the wrong script I think; this isn't triggered in the `_test` CI jobs for this PR. I went off the info in `.circleci/README` about this script being run in all jobs, but that was talking about nightly binary builds I believe. It's a little bit of a jungle.

What I do find in the logs is:
```
Apr 20 21:43:29 + pip_install --user mypy
Apr 20 21:43:29 + pip install --progress-bar off --user mypy
```
but I can't find those lines in either this repo or in `builder`.

So I think this should be removed (and this PR can be merged without it), but where should it go?

Reply:
So I think:
- install in `/.circleci/docker/common/install_travis_python.sh`
- test in test suite rather than a CI file, add to `test/test_type_hints.py`

Reply:
Oops, yes, `binary_linux_test` is only run for binary jobs.